vectorMath.py

```diff
@@ -0,0 +1,79 @@
+import math
+
+
+def magnitude(u):
+    index = 0
+    sum = 0
+    for i in u:
+        mag = (u[index])**2
+        index += 1
+        sum = sum + mag
+    return math.sqrt(sum)
+
+
+def dotProduct(u, v):
+    index = 0
+    sum = 0
+    for i in u:
```

Comment:
You're not using `i` again!

Reply:
This is warmer, but still not the best...
```
def dotProduct(u, v):
    sum = 0
    for i in xrange(len(u)):
        dot = (u[i] * v[i])
        sum = sum + dot
    return sum
```

Reply:
warmer...
```
def dotProduct(u, v):
    sum = 0
    for i, j in zip(u, v):
        dot = i * j
        sum = sum + dot
    return sum
```

Reply:
what is zip?

Reply:
bam...
```
def dotProduct(u, v):
    return sum(i * j for i, j in zip(u, v))
```

Reply:
whoa...

Reply:
This is a "comprehension" how could we do it with lambda?

Reply:
lambda i, j : i*j, u, v

does that work?

Reply:
Try it. You tell me.

Reply:
Also, as for zip. Open a terminal and just type `python` Then, try it. You tell me.

Reply:
it does not work

Reply:
💄

Reply:
New and improved! Now with better variable names!
```
def dotProduct(vector_1, vector2_):
    return sum(
        element_1 * element_2
         for element_1, element_2 in zip(vector_1, vector_2)
)
```